h1st/model/fuzzy/fuzzy_logic_rules.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from typing import NoReturn
+
+import skfuzzy
+from skfuzzy import control as skctrl
+from skfuzzy.control.term import Term
+
+from h1st.model.fuzzy.enums import FuzzyMembership as fm
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class FuzzyLogicRules:
+    def __init__(self) -> None:
+        self.membership = {
+            fm.TRIANGLE: skfuzzy.trimf,
+            fm.TRAPEZOID: skfuzzy.trapmf,
+            fm.GAUSSIAN: skfuzzy.gaussmf,
+        }   
+        self.vars = {}
+        self.rules = {}
+
+    def add_rule(self, name, if_: Term, then_: Term): #  -> NoReturn
+        """
+        Add a fuzzy rule. Place antecedent type variables in 'if' statement
+        and place consequent type variables in 'then' statement.
+
+        .. code-block:: python
+            :caption: example
+            flr = FuzzyLogicRules()
+            flr.add_rule(
+                'rule1',
+                if_=flr.vars['sensor1']['abnormal'],
+                then_=flr.vars['problem1']['yes'])
+        """
+        self.rules[name] = skctrl.Rule(if_, then_)
+
+    def remove_rule(self, name: str) -> NoReturn:
+        self.rules.pop(name, None)
+
+    def add_variable(self, range_: list, name: str, membership_funcs: list, type_: str): #  -> NoReturn
```

Comment:
Why is a fuzzy variable being added to a set of fuzzy rules? Think about this question and consider restructuring a bit.